packages/ui/src/elements/DocumentDrawer/index.tsx

```diff
@@ -64,17 +52,11 @@ export const DocumentDrawer: React.FC<DocumentDrawerProps> = (props) => {
 }
 
 export const useDocumentDrawer: UseDocumentDrawer = ({ id, collectionSlug }) => {
-  const drawerDepth = useEditDepth()
-  const uuid = useId()
+  const drawerDepth = useDrawerDepth()
```

Comment:
Should this continue to use edit depth? If a nested document is opened without a drawer, it needs to still increment the slug.

Reply:
What slug are you referring to here?

Reply:
The slug of the drawer. Theoretically, if you rendered a document directly within a document drawer, then tried to open _another_ drawer within that, the slug of that drawer would not properly reflect the new edit depth.

Reply:
If you want edit depth you would subscribe the useEditDepth.

But drawerDepth and editDepth are both self incrementing, so I'm not following; I think the drawer slugs would be correct 🤔

Reply:
@jacobsfletch I think I see what you are saying, but since it is using the `useDrawerSlug` the slugs should be unique.

Drawer content is wrapped with a DrawerDepthProvider. So nested drawers will have unique slugs via depth. Only 1 drawer at a certain depth should be rendered at a time.

I don't think they would conflict since we are formatting drawer slugs with [`useDrawerSlug`](https://github.dev/payloadcms/payload/tree/chore/beta/depth-providers) which uses a unique identifier, the base slug and the drawerDepth to generate a truly unique drawer slug.

Reply:
But we don't want to rely on `useId` to make that slug unique.

> Should this continue to use edit depth? If a nested document is opened without a drawer, it needs to still increment the slug.

Here's a better example:

```tsx
<EditView>
    <DocumentFields />
    <DocumentDrawer> // This drawer needs an edit depth of 1, drawer depth of 1
        <DocumentDrawer /> // This drawer needs an edit depth of 2, drawer depth of 2
    </DocumentDrawer>
    <DocumentInfoProvider>
        <DocumentFields />
        <DocumentDrawer> // This drawer needs an edit depth of 2, drawer depth of 1
            <DocumentDrawer /> // This drawer needs an edit depth of 2, drawer depth of 2
        </DocumentDrawer>
    <DocumentInfoProvider>
</EditView>
```

If this component reads drawer depth instead of edit depth, then these drawers would technically have the same slug (minus the unique id).

Reply:
Not sure if you saw my comment on the other thread, but wouldn't this be true today if you had 2 relationship fields referencing the same document inside a form? useId would be the only thing making them unique.

Reply:
That is true but at least their drawer and edit depth indices would be accurate

Reply:
So it sounds like you would want three providers?
- EditDepthProvider (useful for the document drawer component)
- FormDepthProvider (useful for unique field labels/id's inside nested forms, forms do not need to be rendered int he context of a document so cannot rely on EditDepth)
- DrawerDepthProvider (useful for making the drawers agnostic and not tied to anything payload specific)

Is that accurate? I can implement this if you feel like that is what's needed. I just want to decouple drawers from edit depth as they are not related. In the DocumentDrawer I can see how they would be related.